sdv/metadata/multi_table.py

```diff
@@ -446,6 +451,29 @@ def validate(self):
                 'The metadata is not valid' + '\n'.join(str(e) for e in errors)
             )
 
+    def add_table(self, table_name):
+        """Add a table to the metadata.
+
+        Args:
+            table_name (str):
+                The name of the table to add to the metadata.
+
+        Raises:
+            Raises ``ValueError`` if ``table_name`` is not valid.
+        """
+        if not table_name:
```

Comment:
I think that we can check if this is `str` otherwise you can asign an object to the dictionary.

Reply:
I think that I would do something like this:
```python
if not isinstance(table_name, str) or not table_name:
    table_name = table_name or "''"
    raise ValueError(
           f"Cannot add a table named '{table_name}' because it already exists in the "
           'metadata. Please choose a different name.'
    )
```

Otherwise this happens:
```
In [15]: mtm = MultiTableMetadata()

In [16]: mtm.add_table(object())

In [17]: mtm
Out[17]: ---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/.virtualenvs/SDV.review/lib/python3.8/site-packages/IPython/core/formatters.py:707, in PlainTextFormatter.__call__(self, obj)
    700 stream = StringIO()
    701 printer = pretty.RepresentationPrinter(stream, self.verbose,
    702     self.max_width, self.newline,
    703     max_seq_length=self.max_seq_length,
    704     singleton_pprinters=self.singleton_printers,
    705     type_pprinters=self.type_printers,
    706     deferred_pprinters=self.deferred_printers)
--> 707 printer.pretty(obj)
    708 printer.flush()
    709 return stream.getvalue()

File ~/.virtualenvs/SDV.review/lib/python3.8/site-packages/IPython/lib/pretty.py:410, in RepresentationPrinter.pretty(self, obj)
    407                         return meth(obj, self, cycle)
    408                 if cls is not object \
    409                         and callable(cls.__dict__.get('__repr__')):
--> 410                     return _repr_pprint(obj, self, cycle)
    412     return _default_pprint(obj, self, cycle)
    413 finally:

File ~/.virtualenvs/SDV.review/lib/python3.8/site-packages/IPython/lib/pretty.py:778, in _repr_pprint(obj, p, cycle)
    776 """A pprint that just redirects to the normal repr function."""
    777 # Find newlines and replace them with p.break_()
--> 778 output = repr(obj)
    779 lines = output.splitlines()
    780 with p.group():

File ~/Projects/SDV/SDV.review/sdv/metadata/multi_table.py:613, in MultiTableMetadata.__repr__(self)
    611 def __repr__(self):
    612     """Pretty print the ``MultiTableMetadata``."""
--> 613     printed = json.dumps(self.to_dict(), indent=4)
    614     return printed

File /usr/lib/python3.8/json/__init__.py:234, in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    232 if cls is None:
    233     cls = JSONEncoder
--> 234 return cls(
    235     skipkeys=skipkeys, ensure_ascii=ensure_ascii,
    236     check_circular=check_circular, allow_nan=allow_nan, indent=indent,
    237     separators=separators, default=default, sort_keys=sort_keys,
    238     **kw).encode(obj)

File /usr/lib/python3.8/json/encoder.py:201, in JSONEncoder.encode(self, o)
    199 chunks = self.iterencode(o, _one_shot=True)
    200 if not isinstance(chunks, (list, tuple)):
--> 201     chunks = list(chunks)
    202 return ''.join(chunks)

File /usr/lib/python3.8/json/encoder.py:431, in _make_iterencode.<locals>._iterencode(o, _current_indent_level)
    429     yield from _iterencode_list(o, _current_indent_level)
    430 elif isinstance(o, dict):
--> 431     yield from _iterencode_dict(o, _current_indent_level)
    432 else:
    433     if markers is not None:

File /usr/lib/python3.8/json/encoder.py:405, in _make_iterencode.<locals>._iterencode_dict(dct, _current_indent_level)
    403         else:
    404             chunks = _iterencode(value, _current_indent_level)
--> 405         yield from chunks
    406 if newline_indent is not None:
    407     _current_indent_level -= 1

File /usr/lib/python3.8/json/encoder.py:376, in _make_iterencode.<locals>._iterencode_dict(dct, _current_indent_level)
    374     continue
    375 else:
--> 376     raise TypeError(f'keys must be str, int, float, bool or None, '
    377                     f'not {key.__class__.__name__}')
    378 if first:
    379     first = False

TypeError: keys must be str, int, float, bool or None, not object
```